tests/core/test_variable.py

```diff
@@ -0,0 +1,558 @@
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+        #test creation with only kwvars
+        self.assertCountEqual(w.params, ('t','u','v'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+        #test creation with vardicts and kwvars
+        w = DepVar({'t':t, 'u':u}, v=v)
+        self.assertCountEqual(w.params, ('t','u','v'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+    def test_dependency_graph(self):
+        """Test dependency_graph method."""
+        t = relentless.DesignVariable(value=1.0)
+        u = relentless.DesignVariable(value=2.0)
+        v = relentless.DesignVariable(value=3.0)
+
+        #test 1 level of dependency
+        w = DepVar(t=t, u=u, v=v)
+        g = w.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v']])
+
+        #test 2 levels of dependency
+        x = DepVar(t=t, v=v)
+        y = DepVar(w=w, x=x)
+        g = y.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v'],['t'],['v'],['w'],['x']])
+
+        #test more complex dependency
+        z = DepVar(t=t, w=w, y=y)
+        g = z.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y,z])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x),
+                                        (z,t),(z,w),(z,y)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v'],['t'],['v'],['w'],['x'],['t'],['w'],['y']])
+
+        #test 'multiple' dependency on same object
+        q = DepVar(t=t, u=t, v=t)
+        g = q.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,q])
+        self.assertCountEqual(g.edges, [(q,t)])
+        params = [g.edges[a,b]['params'] for (a,b) in g.edges][0]
+        self.assertCountEqual(params, ['t','u','v'])
+
+        q = DepVar(t=t, u=t, v=v, w=v)
+        g = q.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,v,q])
+        self.assertCountEqual(g.edges, [(q,t),(q,v)])
+        params0 = sorted([g.edges[a,b]['params'] for (a,b) in g.edges][0])
+        params1 = sorted([g.edges[a,b]['params'] for (a,b) in g.edges][1])
+        self.assertCountEqual([params0,params1], [['t','u'],['v','w']])
+
+        #test circular dependencies
+        a = DepVar(t=t)
+        a.t = DepVar(t=a)
```

Comment:
I got a little confused about this test at first, so to make it more explicit why this is circular, I would suggest:
```suggestion
        a = DepVar(t=t)
        b = DepVar(t=a)
        a.t = b
```
Then use `b` below instead of `a.t`.